src/core/audio/jack_impl.cpp

```diff
@@ -131,19 +131,17 @@ namespace otto::audio {
       auto inputs  = findPorts(JackPortIsPhysical | JackPortIsOutput);
       auto outputs = findPorts(JackPortIsPhysical | JackPortIsInput);
 
-      if (inputs.empty()) {
-        throw global::exception(global::ErrorCode::audio_error,
-          "Couldn't find physical input port");
-      }
       if (outputs.empty()) {
         throw global::exception(global::ErrorCode::audio_error,
           "Couldn't find physical output ports");
       }
 
       bool s;
 
-      s = connectPorts(jack_port_name(ports.input), inputs[0]);
-      if (!s) {global::exit(global::ErrorCode::audio_error); return;}
+      if (!inputs.empty()) {
+        s = connectPorts(jack_port_name(ports.input), inputs[0]);
+        if (!s) {global::exit(global::ErrorCode::audio_error); return;}
```

Comment:
I believe you might be misunderstanding -- the app works fine without an audio input, but it was refusing to run without it.  Since the rPi has no audio input (at least, on the rPi 3), I changed the logic to not *require* the audio input.

Reply:
oh yeah, you just kept my exit call - still, throwing an exit_exception might be better